File: app/src/main/java/com/vinylapp/data/model/Vinyl.kt
package com.vinylapp.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "vinyls")
data class Vinyl(
    @PrimaryKey
    val id: String,
    val title: String,
    val artist: String,
    val year: Int,
    val genre: String,
    val label: String,
    val coverUrl: String,
    val tracklist: List<String>,
    val notes: String = ""
) 

File: src/types/vinyl.ts
export interface Vinyl {
  id: string;
  title: string;
  artist: string;
  year: number;
  genre: string[];
  format: string;
  label: string;
  coverImage: string;
  tracklist: Track[];
  notes?: string;
  condition: 'Mint' | 'Near Mint' | 'Very Good Plus' | 'Very Good' | 'Good' | 'Fair' | 'Poor';
  purchaseDate?: string;
  purchasePrice?: number;
}

export interface Track {
  position: string;
  title: string;
  duration: string;
}

export interface SearchResult {
  id: string;
  title: string;
  artist: string;
  year: number;
  coverImage: string;
  format: string;
}

export interface DiscogsResponse {
  id: string;
  title: string;
  artist: string;
  year: number;
  genre: string[];
  label: string[];
  tracklist: {
    position: string;
    title: string;
  }[];
  images: {
    uri: string;
    height: number;
    width: number;
  }[];
} 